### PR TITLE
[Messenger] Use Doctrine DBAL new Types::* constants

### PR DESCRIPTION
**Pull request by fancyweb**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

### CODE REVIEW

**Comment by ogizanagi** [comment]
This one should target 4.4. 4.3 reached end of bug fixes.

**Comment by fancyweb** [comment]
Thanks, I forgot about that 😅

File: src/Symfony/Component/Messenger/Transport/Doctrine/Connection.php [hunk 1]
```diff
@@ -119,6 +120,7 @@ public function send(string $body, array $headers, int $delay = 0): string
                 'available_at' => '?',
             ]);
 
+        $useDeprecatedConstants = !class_exists(Types::class);
```
Comment (Tobion) [inline src/Symfony/Component/Messenger/Transport/Doctrine/Connection.php]: wouldn't it be better to use a static property initialized once as this triggers the class loader for every send and get (even twice)

**Comment by OskarStark** [comment]
You should update the target branch in the Pr header table as well?

File: src/Symfony/Component/Messenger/Transport/Doctrine/Connection.php (hunk 1 above)
Comment (nicolas-grekas) [reply src/Symfony/Component/Messenger/Transport/Doctrine/Connection.php]: `self::$useDeprecatedConstants = self::$useDeprecatedConstants ?? class_exists(Types::class);` in the constructor (same in the other PR then)

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @fancyweb.

**Pull request merged**